train_segmentation.py

```diff
@@ -792,6 +794,19 @@ def train(cfg: DictConfig) -> None:
 
         cur_elapsed = time.time() - since
         # logging.info(f'\nCurrent elapsed time {cur_elapsed // 60:.0f}m {cur_elapsed % 60:.0f}s')
+    
+    # Script model
+    if scriptmodel:
+        model_to_script = ScriptModel(model,
+                                      device=device,
+                                      input_shape=(1, num_bands, patches_size, patches_size),
+                                      mean=mean,
+                                      std=std,
+                                      min=scale[0],
+                                      max=scale[1])
+        
+        scripted_model = torch.jit.script(model_to_script)
+        scripted_model.save(output_path.joinpath('scripted_model.pt'))
```

Comment:
The scripting logic is after all training and validation loops are completed. This means scripting and saving the resulting model is done just once. Also, the whole scripting logic is enabled by choice with a flag; hence you can run it without training your model. 

On the generic name, I did not give it much thought; any suggestions we can explore? One solution is to utilize the naming convention for the raw weights that's saved at latest best validation loss.

Reply:
Thanks for the clarifications. The naming is only important if it will be shared outside of the output_path. I think we can look at it later if we see it necessary.